pkg/collectd/collectd.go

```diff
@@ -7,7 +7,7 @@ import (
 	jsoniter "github.com/json-iterator/go"
 )
 
-type CollectdMeta struct {
+type collectdMeta struct {
```

Comment:
👍

The performance of this should probably be analysed more in depth. Golang object allocation can be expensive time-wise. This is not for this PR but I will some testing for the SG core refactor

Reply:
I wonder if there is some profile tooling we could use to help us analyse these types of code paths?

...there is... I did some digging, and here are a few things we can look in to as part of the refactoring:

* https://goreportcard.com/report/github.com/infrawatch/sg-core
* https://blog.golang.org/pprof
* https://github.com/golangci/golangci-lint